uktrade/data-hub-frontend
# Feature/filter by investment type

## Description of change

* Adds filter by investment type checkbox to the investment projects page
* Adds tests for the stage checkbox filters
* Refactors the `RoutedCheckboxes` component into `CheckboxGroupField` and `RoutedCheckboxField` to simplify them (and following the Single Responsibility Principal)
* Refactors the `FilterAdvisersTypeahead`, which is now `RoutedAdvisersTypeahead` - this wraps the existing `RoutedTypeahead` to avoid code duplication. The new typeahead gets the adviser ids from the `selectedFilters` object instead of props (this is now consistent with how the chips work)
* Refactors the investments filter tests - moves out common assertions and helpers to support functions
* Refactors `RoutedDownloadDataHeader` into two components including a lower-level `DownloadDataHeader` that does not depend on routing as well as `RoutedDownloadDataHeader` which wraps it.
* Other small bits of refactoring including removing props which were no longer used or needed from filter components

## Test instructions

There should now be a set of checkboxes to filter by investment type underneath UK region. If you select an option, it should appear as a chip in the list header and also in the url. Clearing the chip or selecting clear all should uncheck the option. The other filters should also continue to operate as normal.

## Screenshots

![Screenshot from 2020-12-14 19-21-46](https://user-images.githubusercontent.com/1234577/102125493-b2e3c500-3e41-11eb-9cd3-f52af44d844a.png)

## Checklist

[//]: # "When submitting a PR make sure the code review guidelines have been satisfied.
https://github.com/uktrade/data-hub-frontend/blob/master/docs/Code%20review%20guidelines.md"
